### PR TITLE
feat(suggest): suggest.asciiCharactersOnly option

### LINKED ISSUE #1459: Provide an option to complete for ASCII-only characters
**Is your feature request related to a problem? Please describe.**

I was wondering if there is a way to complete when typing ASCII characters, and don't complete for a non-ASCII prefix. If not, could you add an option to achieve this?

Here is the story. I'm a Chinese user. When I'm writing Chinese, the bundled sources (`around`, `buffer` and `file`) will do the completion and show some sentences matching the prefix that I have typed. This is really annoying. Because completing words for Chinese is somewhat unrealistic, I prefer that the coc.nvim could ignore them and don't show these results.

**Describe the solution you'd like**

I don't know if there is an existing way to handle this problem. If there is, please let me know.

So my suggestion is to filter the completion results and test whether they contain any non-ASCII characters. Provide an option to configure this behavior, e.g. `completeForAsciiCharactersOnly`.

Or more generally, could there be a mechanism to define some user rules to filter completion results? Just out of curiosity.

### PR DESCRIPTION
**Pull request by fannheyward**
close #1459

**default false**

~cc @chemzqm I make `suggest.asciiCharactersOnly` default true, any suggestions?~

### CODE REVIEW

File: src/completion/index.ts [hunk 1]
```diff
@@ -144,6 +144,7 @@ export class Completion implements Disposable {
       localityBonus: getConfig<boolean>('localityBonus', true),
       highPrioritySourceLimit: getConfig<number>('highPrioritySourceLimit', null),
       lowPrioritySourceLimit: getConfig<number>('lowPrioritySourceLimit', null),
+      asciiCharactersOnly: getConfig<boolean>('asciiCharactersOnly', false)
```
Comment (voldikss) [inline src/completion/index.ts]: The default value is `true` in `schema.json` but  `false` here. could be a forgetten work

**Comment by fannheyward** [comment]
@voldikss bingo, fixed

**Comment by chemzqm** [comment]
@fannheyward to not break current behavior, It should be false by default.

**Comment by fannheyward** [comment]
Make default as false.

**Pull request merged**